lib/reservation.rb

```diff
@@ -0,0 +1,19 @@
+require 'date'
+
+module BookingSystem
+  class Reservation
+    attr_reader :name, :check_in, :check_out, :room_num, :total_cost
+
+    COST_PER_NIGHT = 200
+
+    def initialize(name, room_num, check_in, check_out)
+      dates = DateRange.new(check_in, check_out)
+      @name = name
+      @check_in = dates.check_in
+      @check_out = dates.check_out
```

Comment:
If `Reservation` has a `DateRange`, you should not also be storing the checkin and checkout dates directly. Instead, anyone who wants the checkin or checkout date should have to go through the `DateRange` object.